juju/juju
# Extract s3 client and session into their package and create a purpose-built charm s3 client

We are soon to start using the s3 client in the Juju client itself. So extract the client definitions into a new 'internal' directory, so it can be accessed across Juju

Also create a charmclient wrapper around the s3client. This is very lightweight at the moment, since we only implement GetObject, which doesn't require anything special to get charms. This will be required in future when we start doing more complex things driving the s3client

Also, drop agentConfig from S3 client constructer since it isn't used and replace logger interface with concrete loggo.Logger

## Checklist


## QA steps

Verify workers can still use the s3 client by bootstrapping and deploying a charm
```
juju bootstrap lxd lxd
juju add-model m
juju deploy ubuntu
```
